DSharpPlus/Entities/Interaction/Components/DiscordActionRowComponent.cs

```diff
@@ -38,19 +38,17 @@ public sealed class DiscordActionRowComponent : DiscordComponent
         /// The type of component this represents. Always returns type 1.
         /// </summary>
         [JsonProperty("type", NullValueHandling = NullValueHandling.Ignore)]
-        public ComponentType Type { get; internal set; } = ComponentType.ActionRow;
+        internal ComponentType Type { get; set; } = ComponentType.ActionRow;
 
         /// <summary>
         /// The components contained within the action row.
         /// </summary>
         [JsonProperty("components", NullValueHandling = NullValueHandling.Ignore)]
-        public ReadOnlyCollection<DiscordComponent> Components { get; internal set; }
-
-        internal DiscordActionRowComponent() { }
+        public IReadOnlyCollection<DiscordComponent> Components { get; internal set; }
 
         internal DiscordActionRowComponent(IEnumerable<DiscordComponent> components)
         {
-            this.Components = components.ToList().AsReadOnly();
+            this.Components = components.ToArray();
```

Comment:
akshually this is mutable

Reply:
Ack. Yeah, I was thinking about how this should be handled since someone may want to remove a component after the fact, and then they'd have to new up a new action row. Bleh. 

I forget who told me to make the components immutable.